src/Equinor.ProCoSys.IPO.Command/InvitationCommands/SignPunchOut/SignPunchOutCommandValidator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Command.Validators.InvitationValidators;
+using Equinor.ProCoSys.IPO.Command.Validators.RowVersionValidators;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using FluentValidation;
+
+namespace Equinor.ProCoSys.IPO.Command.InvitationCommands.SignPunchOut
+{
+    public class SignPunchOutCommandValidator : AbstractValidator<SignPunchOutCommand>
+    {
+        public SignPunchOutCommandValidator(IInvitationValidator invitationValidator, IRowVersionValidator rowVersionValidator)
+        {
+            CascadeMode = CascadeMode.Stop;
+
+            RuleFor(command => command)
+                .MustAsync((command, token) => BeAnExistingInvitation(command.InvitationId, token))
+                .WithMessage(command =>
+                    $"IPO with this ID does not exist! Id={command.InvitationId}")
+                .MustAsync((command, token) => BeAnExistingParticipant(command.ParticipantId, command.InvitationId, token))
+                .WithMessage(command =>
+                    $"Participant with this ID does not exist! Id={command.ParticipantId}")
+                .MustAsync((command, token) => BeANonCanceledInvitation(command.InvitationId, token))
+                .WithMessage(command =>
+                    "Invitation is canceled, and thus cannot be signed!")
+                .Must(command => HaveAValidRowVersion(command.ParticipantRowVersion))
+                .WithMessage(command =>
+                    $"Participant row version is not valid! ParticipantRowVersion={command.ParticipantRowVersion}")
+                .MustAsync((command, token) => BeASigningParticipantOnIpo(command.InvitationId, command.ParticipantId, token))
+                .WithMessage(command =>
+                    $"The IPO does not have a participant assigned to sign the IPO with this ID! ParticipantId={command.ParticipantId}")
```

Comment:
Was a little confused by this. Suggestion "Participant is not assigned to sign this IPO! ParticipantId={}"